External/Plugins/HaXeContext/Linters/DiagnosticsLinter.cs

```diff
@@ -0,0 +1,77 @@
+﻿using LintingHelper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ASCompletion.Completion;
+using ASCompletion.Context;
+using PluginCore;
+using PluginCore.Managers;
+
+namespace HaXeContext.Linters
+{
+    class DiagnosticsLinter : ILintProvider
+    {
+        private static Dictionary<HaxeDiagnosticsSeverity, LintingSeverity> conversionDict =
+            new Dictionary<HaxeDiagnosticsSeverity, LintingSeverity>()
+            {
+                {HaxeDiagnosticsSeverity.ERROR, LintingSeverity.Error},
+                {HaxeDiagnosticsSeverity.INFO, LintingSeverity.Info},
+                {HaxeDiagnosticsSeverity.WARNING, LintingSeverity.Warning}
+            };
+
+        public void LintAsync(string[] files, LintCallback callback)
```

Comment:
@chipshort Before using this functionality, you need to check the compiler version to convince that "@diagnostics" is available